activestorage/test/models/attachment_test.rb

```diff
@@ -134,6 +134,15 @@ class ActiveStorage::AttachmentTest < ActiveSupport::TestCase
     assert_equal blob, ActiveStorage::Blob.find_signed(signed_id)
   end
 
+  test "can destroy attachment without existing relation" do
+    blob = create_blob
+    @user.highlights.attach(blob)
+    attachment = @user.highlights.find_by(blob_id: blob.id)
+    attachment.update_attribute(:name, "old_highlights")
+    assert_nil attachment.send(:dependent)
```

Comment:
```suggestion
```

Reply:
I don't think this test adds any value, we generally don't care about the result of calling private APIs directly

Reply:
I was showing the result was not purge_later but rather nil. Since this is the behavior. If later we want want to change the default behavior to purge_later for attachments without a record relationship, then this test would fail and need to be updated. 

I can remove it, but I was just making sure it would fail later if we changed that assumption, but to your point that could be tested a different way and not via the private API here.